packages/build/src/build/buildAuth/validateAuthConfig.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable no-param-reassign */
+
+/*
+  Copyright 2020-2022 Lowdefy, Inc
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import { type } from '@lowdefy/helpers';
+import { validate } from '@lowdefy/ajv';
+import lowdefySchema from '../../lowdefySchema.js';
+
+async function validateAuthConfig({ components }) {
+  if (type.isNone(components.auth)) {
+    components.auth = {};
+  }
+  if (!type.isObject(components.auth)) {
+    throw new Error('lowdefy.auth is not an object.');
+  }
+  if (type.isNone(components.auth.pages)) {
+    components.auth.pages = {};
+  }
+  if (type.isNone(components.auth.pages.roles)) {
+    components.auth.pages.roles = {};
+  }
+  if (type.isNone(components.auth.providers)) {
+    components.auth.providers = [];
+  }
+
+  const { valid } = validate({
+    schema: lowdefySchema.definitions.authConfig,
+    data: components.auth,
+    returnErrors: true,
+  });
+
+  if (!valid) {
+    throw new Error('lowdefy.auth does not match schema.'); // TODO: Better error message
```

Comment:
validate returns the error messages defined in the schema.

Reply:
Aren't we doing this test in testSchema?

Reply:
We have already logged all the schema errors as warnings, so not sure if we should log them again.